### ISSUE TITLE
Bootloop when booting with 8-Port SATA Adapter

### ISSUE THREAD

**Issue by Ryeera** [AuxXxilium/arc#832]
**Describe the Bug**
I built a server in a Jonsbo N3 case which allows 8 SATA-HDDs to be installed. I then installed Proxmox on it and now wanted to virtualize a Synology NAS on it to replace my current DS1815+. So I pass through the SATA-Card, which is a KALEA-INFORMATIQUE M.2 card which has two MiniSAS-Ports that break out to 8 SATA-connectors. It is using the JMB585 and JMB575 chipsets. I assume this might be a problem, but I'm not really well versed when it comes to this stuff so I hope someone can help me.
When I then boot into the Arc Bootloader, it all runs fine. I build the bootloader and reboot but when it finishes, it shortly shows the screen telling me to connect to the NAS via port 5000, but after just a few seconds, it suddenly reboots.
The 8 harddrives are correctly detected within the Arc-Bootloader and I can format all of them just fine, but when I try to boot into DSM, it crashes and reboots.
I tried this with a 6-Port SATA-Card first and this worked fine when passing it through to the VM. I'm very close to just giving up on the extra 2 ports and settling for 6. Which honestly would probably be enough.

I'm sorry if I'm being dumb ^^"

**Screenshots**
Sysinfo-Code: BPCUZ7BJQ

**Hardware**
CPU: AMD Ryzen 5 4650G
Mainboard: GigaByte B550I Aorus Pro AX
RAM: 2x KTH-PL432E/32G
Network-Card: RTL8125 2.5 GBit/s
Storage -Controller: JMB58x AHCI SATA Controller (https://www.amazon.de/dp/B0BDMSZR41?psc=1&ref=ppx_yo2ov_dt_b_product_details)

**Comment by Ryeera** [comment]
<img width="433" alt="image" src="https://github.com/AuxXxilium/arc/assets/79052127/07e2247e-1276-4204-80a6-ceebcc9cfdef">
Just noticed that the JMB58x Controller only shows with 5 drives in the sysinfo screen, it should have 8.

**Comment by AuxXxilium** [comment]
than you use an flashed chinese grap controller. these cheap controller are ...

if it reboots, it run in kernel panic, this is caused by your hardware and dsm can't work with it.

**Comment by AuxXxilium** [comment]
first you should reset loader and don't select any additional addon.

**Comment by Ryeera** [comment]
> first you should reset loader and don't select any additional addon.

Alright I will try this, one second!


> than you use an flashed chinese grap controller. these cheap controller are ...
> 
> if it reboots, it run in kernel panic, this is caused by your hardware and dsm can't work with it.

That would be really unfortunate as this is literally the ONLY M.2-card which supports 8 SATA-Ports that I could find anywhere... And M.2 is my only choice because I am using a splitter-card that lets me split the one PCIe x16 slot into 4 M.2-Slots.

**Comment by AuxXxilium** [comment]
is your board able to do: bifurcation for pcie?

**Comment by Ryeera** [comment]
> is your board able to do: bifurcation for pcie?

Yeah this is the only way I can use this card. But since the case also doesn't have more space for more cards, I can't use a normal PCIe bifurcation card. And I need more 2.5 GBit ports and more SATA-Ports, so I have to use at least 2 cards. Then plus at least once more M.2-slot for an SSD.

**Comment by Ryeera** [comment]
Unfortunately, still the same issue. I even changed the Bios to UEFI because I noticed it was still on Legacy. However, it didn't work with a brand-new Arc-Loader...
I guess a solution would be to just use the 6-port SATA-adapter I got, pass that through (to have proper HDD-info, especially regarding broken sectors and less performance-loss due to virtualizing the HDDs) and the other two HDDs I can just pipe in through the motherboard's chipset and use the for Backup in RAID 0.

**Comment by AuxXxilium** [comment]
try a broadwellnk model and try again

**Comment by Ryeera** [comment]
> try a broadwellnk model and try again

This actually worked, I could boot into DSM, but it only shows 5 HDDs instead of 8. As it does in the Bootloader. The Portmap it automatically configured it 65. For some reason, it only detects the controller with 5 ports and not 8.

**Comment by AuxXxilium** [comment]
because your controller can't address more than these 5.

**Comment by Ryeera** [comment]
How come I can see and format all 8 connected drives in the Arc-Bootloader then? That's the part that confuses me...

**Comment by AuxXxilium** [comment]
wrong portmap selected

**Comment by Ryeera** [comment]
> wrong portmap selected?

I selected Max Ports. Would I just manually set it to 68?

**Comment by AuxXxilium** [comment]
if i look at your picture, there are only 5 ports on this controller.

**Comment by Ryeera** [comment]
<img width="693" alt="image" src="https://github.com/AuxXxilium/arc/assets/79052127/9af373dd-9379-45cc-92ed-36519f201a47">

What I seriously don't get it that all disks are there. I can even format all of them correctly, so the controller must be able to address them, right? I don't know where Arc gets the information from that the controller only has 5 disks connected because it definitely has 8. There are no other controllers in my system.

**Comment by AuxXxilium** [comment]
this is strange, because script is counting connected and initialized disks, so it looks like they are there but can‘t be initialized.

**Comment by Ryeera** [comment]
Alright thanks a lot! I will debug a little once I'm home from work ^^

**Comment by Ryeera** [comment]
<img width="1076" alt="image" src="https://github.com/AuxXxilium/arc/assets/79052127/43e566b5-4984-4e28-ba9d-487074445740">

Okay I debugged some and have these findings:
1. ls /sys/class/scsi_host does in fact report only 5 devices for the one controller
2. lsscsi however does report all HDDs

I did notice that the last 4 drives have the address 10:x:0:0, where x is between 0 and 3. I assume that means that somehow, this controller connects 4 drives to the same port? Is that what a port-multiplier does maybe? I honestly am not smart enough for this but I'm hoping that you understand what's going on here and if we can even work with this or not. If not, I'm gonna have to resort to plan b ^^"
